### PR TITLE
samples: net: ptp: slave: Add PTP slave sample

### PR DESCRIPTION
**Pull request by alsrgv**
The PTP slave sample application for Zephyr allows boards to synchronize to a IEEE 1588-2008 (PTP V2) master running on the network.

This example will be reworked based on feedback from #40187.

cc @jukkar @erwango

Signed-off-by: Alex Sergeev <asergeev@carbonrobotics.com>

### CODE REVIEW

File: samples/net/ptp/slave/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.13.1)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(ptp_slave)
+
+file(GLOB APP_SOURCES src/*.c)
+file(GLOB APP_HEADERS src/*.h)
```
Comment (SebastianBoe) [inline samples/net/ptp/slave/CMakeLists.txt:7]: adding header files to app is not normal.

Either stop doing this, or add a comment in the build script justifying it.

**Review by SebastianBoe — CHANGES_REQUESTED** [review]
reviewed build system

**Review by SebastianBoe — DISMISSED** [review]
Build and config scripts are OK

File: samples/net/ptp/slave/README.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+The PTP slave sample application for Zephyr allows boards to synchronize
+to a IEEE 1588-2008 (PTP V2) master running on the network.
+
+The source code for this sample application can be found at:
+:zephyr_file:`samples/net/ptp/slave`.
+
+Requirements
+************
+
+This example requires one of the boards listed below and a IEEE 1588-2008
```
Comment (erwango) [inline samples/net/ptp/slave/README.rst:18]: Either you need to add the boards or rework that sentence

File: samples/net/ptp/slave/README.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+on the Linux machine.
+
+Building and Running
+********************
+
+A good way to run this sample is to run this PTP slave application with
+an embedded device like NXP FRDM-K64F, Nucleo-H743-ZI, Nucleo-H745ZI-Q,
+Nucleo-F767ZI or Atmel SAM-E70 Xplained. Note that PTP is only supported for
+boards that have an Ethernet port and which has support for collecting
+timestamps for sent and received Ethernet frames.
```
Comment (erwango) [inline samples/net/ptp/slave/README.rst:31]: Move most part of this under Requirements

File: samples/net/ptp/slave/sample.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+common:
+  harness: net
+  tags: net ptp
+sample:
+  description: Test PTP functionality
+  name: PTP sample app
+tests:
+  sample.net.ptp.slave:
+    platform_allow: frdm_k64f sam_e70_xplained nucleo_f767zi nucleo_h743zi nucleo_h745zi_q_m7
+    depends_on: netif
```
Comment (erwango) [inline samples/net/ptp/slave/sample.yaml:10]: I think you should add build_only=true.
@jukkar what do you think ?

**Review by erwango — CHANGES_REQUESTED** [review]
Minor comments

**Comment by carlescufi** [comment]
@erwango please take another look

**Review by tejlmand — APPROVED** [review]

**Comment by erwango** [comment]
> @erwango please take another look

@carlescufi None of my comments were addressed. @alsrgv Can you provide an updated version ?

**Review by jukkar — CHANGES_REQUESTED** [review]
I do not mind adding a PTP sample but there are some issues with this one that requires some more work. This PR introduces duplicate functions / structs in a sample instead of re-using common PTP defines found in `include/net/gptp.h` file. IMHO it will be super confusing if we have various PTP related things in the system header file and new defines etc. in the sample.
Better would be to try to add relevant things that are missing in `include/net/gptp.h` to that file and possible missing/needed functionality to system provided PTP implementation (if that is applicable).

**Comment by alsrgv** [comment]
Apologies, haven't worked on this for a while. @jukkar, on the other PR you [left a comment](https://github.com/zephyrproject-rtos/zephyr/pull/40187#discussion_r771171723) about using socket APIs instead of net_context, but I'm not sure how to retrieve a timestamp from socket API. I'd like to rework that part before touching others.

I agree about `gptp.h` headers - I was planning to introduce `ptp.h` and refactor most of the structures defined in IEEE 1588 to be in `ptp.h` and used by `gptp.h`.

**Comment by alsrgv** [comment]
Attempting to resurrect this PR.

**Comment by alsrgv** [comment]
I will reopen separate PR.

**Pull request closed without merging**